src/ucp/core/ucp_ep.c

```diff
@@ -796,6 +796,7 @@ ucs_status_t ucp_ep_init_create_wireup(ucp_ep_h ep, unsigned ep_init_flags,
         return status;
     }
 
+    ucp_ep_activate_worker_ifaces(ep);
```

Comment:
can be it done in `ucp_worker_get_ep_config` somehow?

Reply:
`ucp_worker_get_ep_config` gets the configuration. However, it doesn't set the configuration to the endpoint. `ucp_ep_activate_worker_ifaces` increases the number of endpoints using the endpoint configuration. Thus it should be only called when the configuration is set to the endpoint.

Reply:
imo increasing `ep_count` of `ep_config` in `ucp_ep_activate_worker_ifaces` is confusing. Maybe it is worth to split `ucp_ep_activate_worker_ifaces` and increase `ep_count` in `ucp_worker_get_ep_config`?

Reply:
But calling `ucp_worker_get_ep_config` doesn't mean that endpoint will use the configuration. It is not correct to increase the counter inside the `ucp_worker_get_ep_config`.

Reply:
We can do the following instead:
```c
...
    get_ep_config(&cfg_idx);
    set_ep_config(ep, cfg_idx);
...

set_ep_config(ep, cfg_idx)
{
    ep->cfg_idx = cfg_idx;
    increase_cnt(cfg_idx);
    activate_ifaces()
}
```

Reply:
yeah, we can add new function ucp_ep_set_cfg_index(ep, new_cfg_index) - that will deactivate according to the exisitng ep->cfg_index (if it is not NULL) and activate according to the new cfg_index